src/fs_io.cc

```diff
@@ -424,16 +424,8 @@ diskHandleRead(int fd, void *data)
 void
 file_read(int fd, char *buf, int req_len, off_t offset, DRCB * handler, void *client_data)
 {
-    dread_ctrl *ctrl_dat;
     assert(fd >= 0);
-    ctrl_dat = (dread_ctrl *)memAllocate(MEM_DREAD_CTRL);
-    ctrl_dat->fd = fd;
-    ctrl_dat->offset = offset;
-    ctrl_dat->req_len = req_len;
-    ctrl_dat->buf = buf;
-    ctrl_dat->end_of_file = 0;
-    ctrl_dat->handler = handler;
-    ctrl_dat->client_data = cbdataReference(client_data);
+    const auto ctrl_dat = new dread_ctrl(fd, offset, req_len, buf, handler, cbdataReference(client_data));
```

Comment:
I recommend moving cbdataReference() call inside dread_ctrl constructor and placing the matching cbdataReferenceDone() call inside its destructor. Doing so will lower several red flags:

* Official diskHandleRead() code sometimes does not call cbdataReferenceDone() when deleting ctrl_dat object.
* PR code prohibits dread_ctrl object moving or copying, but there is no obvious reason to do that (now) because the class effectively remains a POD.

I do not insist on this change despite recommending it.

Reply:
I had considered it, however whenever we touch `cbdataReference*()` there usually occurs a substantial amount of testing and bug fixing side effects.

I am trying to do this with as close to zero logic change as possible, so the issues you point out are good reasons for me not to do this in this PR.